src/lib/eZ/RichText/Resources/stylesheets/docbook/xhtml5/edit/core.xsl

```diff
@@ -688,9 +688,14 @@
         <xsl:value-of select="@ezxhtml:class"/>
       </xsl:attribute>
     </xsl:if>
+    <xsl:if test="@xml:id">
+      <xsl:attribute name="id">
+        <xsl:value-of select="@xml:id"/>
+      </xsl:attribute>
+    </xsl:if>
     <xsl:if test="@ezxhtml:align">
-      <xsl:attribute name="data-ezalign">
-        <xsl:value-of select="@ezxhtml:align"/>
+      <xsl:attribute name="style">
+        <xsl:value-of select="concat( 'text-align:', @ezxhtml:align, ';' )"/>
```

Comment:
For edit mode it still should be data-ezalign AFAIK. @dew326?

Reply:
Apart from front-end remark, out of curiosity: why changing edit mode output is considered BC break if those attributes weren't stored in the database in the first place?

Reply:
Yes, in alloyEditor we have `data-ezalign`.

Reply:
@alongosz it breaks edit otherwise, that's why I've changed it. Unchanged result:
![image](https://user-images.githubusercontent.com/22300504/94914518-798af280-04ab-11eb-927d-378ea7b378a8.png)

Reply:
That's because in our edit mode we have defined float to ezalign right. https://github.com/ezsystems/ezplatform-richtext/blob/master/src/bundle/Resources/public/scss/_alloyeditor-ez.scss#L1280

Why it is not "broken" when you set this property in the AlloyEditor?

Reply:
@dew326 that's how headings, paragraphs and pre works. They don't utilize {{data-ezalign}}, they operate only on {{style="text-align"}}, they would break as well when using {{data-ezalign}}.

Reply:
> @alongosz apart from the front-end remark, out of curiosity: why changing edit mode output is considered a BC break if those attributes weren't stored in the database in the first place?

Because of 3rd party custom work for OE.

When I looked at this code yesterday it was almost okay. Not following the today's change in the approach.

Reply:
That's true we have both. One is for block aligning and the second one is for text alignment in the block. 
And for the custom styles, AlloyEditor sets the text-align.

Reply:
@dew326 so where exactly is the problem?

1. When we create content and we add alignment for the custom block it sets - `style="text-align`, even without this PR.
2. When we leave `data-ezalign` the alignment won't work anymore when editing. 
3. When we set  `style="text-align` it behaves exactly as when creating such content. Why is this change invalid though as it just mirrors behavior when creating?

Reply:
This change is valid. You should have `text-align` in custom styles and only there as AlloyEditor sets it itself and requires it to work properly.

Reply:
@alongosz It's good that we end up with the right solution here, but FYI edit format is primarily for OE, then secondary to allow imports in html5 as a bonus. So if we have a bug with OE <> format which means we need to change it, then we change it. Same as DB schema, we change *if* we _have_ to.

Reply:
> This change is valid. You should have `text-align` in custom styles and only there as AlloyEditor sets it itself and requires it to work properly.

So now the issue is that Custom Tags stopped working (we've just tested it with @dew326), because the change applied here actually covers both Custom Tags and Custom Styles, since till now they were semantically almost the same for Docbook. `data-ezalign` needs to be kept on Custom Tags, while the current approach is acceptable for custom styles.

> @alongosz It's good that we end up with the right solution here  but FYI edit format is primarily for OE, then secondary to allow imports in html5 as a bonus. So if we have a bug with OE <> format which means we need to change it, then we change it. Same as DB schema, we change _if_ we _have_ to.

Well, ok :+1:  Just don't say I didn't warn you :upside_down_face:  Hence I'll accept the solution as long as it doesn't break Custom Tags.

Reply:
We will aim to not break custom tags, I didn't mean we should break that. anyway case closed :)